src/Mvc/Mvc.Core/src/Formatters/TranscodingReadStream.cs

```diff
@@ -6,19 +6,20 @@
 using System.Diagnostics;
 using System.IO;
 using System.Text;
+using System.Text.Unicode;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.AspNetCore.Mvc.Formatters.Json
 {
     internal sealed class TranscodingReadStream : Stream
     {
+        private const int OverflowBufferSize = 4; // The most number of bytes used to represent a single UTF char
```

Comment:
@GrabYourPitchforks I know you said 3, Wikipedia suggests 4. What gives?

Reply:
Is there a super good reason to use this over GetMaxByteCount?

Reply:
It also doesn't matter if it's 4 becuase the minimum sized buffer that will be returned is like at least 16 elements.

Reply:
> Is there a super good reason to use this over GetMaxByteCount?

@GrabYourPitchforks said `GetMaxByteCount` over-estimates and recommended hard-coding a value. But as you pointed out we're almost certain to get a much larger buffer.